File: medium/0034_find_first_and_last_position_of_element_in_sorted_array/solution.cpp
#include <algorithm>
#include <ranges>
#include <vector>

class Solution {
public:
  std::vector<int> searchRange(std::vector<int> &nums, int target) {
    auto [start, end] = std::ranges::equal_range(nums, target);

    if (start == nums.end() || *start != target)
      return {-1, -1};

    return {static_cast<int>(start - nums.begin()),
            static_cast<int>(end - nums.begin()) - 1};
  }
};

File: medium/0034_find_first_and_last_position_of_element_in_sorted_array/solution.py
from bisect import bisect_left, bisect_right


class Solution:
    def searchRange(self, nums: list[int], target: int) -> list[int]:
        start = bisect_left(nums, target)

        if start == len(nums) or nums[start] != target:
            return [-1, -1]

        return [start, bisect_right(nums, target) - 1]
